lib/tapioca/gem/listeners/methods.rb

```diff
@@ -142,6 +142,35 @@ def compile_method(tree, symbol_name, constant, method, visibility = RBI::Public
           tree << rbi_method
         end
 
+        sig { params(method: UnboundMethod, constant: Module).returns(T::Boolean) }
+        def method_defined_by_constant?(method, constant)
+          # Check whether the method is defined on the constant.
+          #
+          # In most cases, it works to check that the constant is the method owner. However,
+          # in the case that a method is also defined in a module prepended to the constant, it
+          # will be owned by the prepended module, not the constant.
+          #
+          # This method implements a better way of checking whether a constant defines a method.
+          # It walks up the ancestor tree via the `super_method` method; if any of the super
+          # methods are owned by the constant, it means that the constant declares the method.
```

Comment:
Could we have this comment on top of the method as documentation rather than inside?